GVFS/GVFS.UnitTests/Common/GitStatusCacheTests.cs

```diff
@@ -154,5 +154,31 @@ public void CanRefreshCache()
                 statusCache.Shutdown();
             }
         }
+
+        [TestCase]
+        public void HandlesExceptionCreatingDirectory()

```

Comment:
Do we need to test different exception types since we are catching two different types `IOException` and `UnauthorizedAccessException` and all others would not be handled?

Reply:
Interesting question - I think we could test different exceptions, and it would help verify the types of exceptions we are capturing. As is, we test that the exceptions we handle are being handled correctly (which I think provided a good bit of value on its own).

Reply:
> As is, we test that the exceptions we handle are being handled correctly

The code also handles `UnauthorizedAccessException` should we test that it's being handled correctly (as Kevin suggested)?

One might be to change `CreateDirectoryThrowsException` to something like ` Exception ExceptionThrownByDirectory { get; set; }` and allow the test to set the specific exception it wants `CreateDirectory` to throw.